sources/light_client.move

```diff
@@ -401,16 +401,6 @@ public fun calc_next_required_difficulty(lc: &LightClient, parent_block: &LightB
 
     // if this block does not start a new retarget cycle
     if ((parent_block.height() + 1) % blocks_pre_retarget != 0) {
-        if (params.adjust_difficulty()) {
```

Comment:
During the PP session, we designed it in the following way:
- we have `params.skip_difficulty_check()` to see if we should validate difficulty. 
- we have `adjust_difficulty` which we use in regtest.